contracts/ENSGovernorV2.sol

```diff
@@ -0,0 +1,159 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.2;
+
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "./ENSGovernor.sol";
+
+/**
+ * @title ENSGovernorV2
+ * @dev Enhanced ENS Governor with bonding functionality.
+ */
+contract ENSGovernorV2 is ENSGovernor {
+    using SafeERC20 for ERC20Votes;
+
+    uint256 public forfeitedBondsBalance;
+    uint256 public lockedBondsBalance;
+
+    /**
+     * @dev ProposalBond represents the bond associated with a proposal.
+     */
+    struct ProposalBond {
+        uint256 amount; // The amount of bond deposited.
+        address proposer; // The address of the proposer who deposited the bond.
+        bool refunded; // Flag indicating if the bond has been refunded.
+        bool forfeited; // Flag indicating if the bond is forfeited.
```

Comment:
Nit: it would be a bit cleaner and gas efficient to implement this with a single status variable with three enumerated statuses (pending, refunded, forfeited), rather than two bools.